tensorflow_probability/python/distributions/pert.py

```diff
@@ -0,0 +1,164 @@
+               allow_nan_stats=False,
+               name="PERT"):
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      dtype = dtype_util.common_dtype([mini, maxi, mode], tf.float32)
+      self._min = tensor_util.convert_nonref_to_tensor(
+          mini, name='mini', dtype=dtype)
+      self._mod = tensor_util.convert_nonref_to_tensor(
+          mode, name='mode', dtype=dtype)
+      self._max = tensor_util.convert_nonref_to_tensor(
+          maxi, name='maxi', dtype=dtype)
+      self._smoothness = tensor_util.convert_nonref_to_tensor(
+          smoothness, name='smoothness', dtype=dtype)
+      self._concentration1 = 1. + self._smoothness \
+        * (self._mod - self._min) / (self._max - self._min)
+      self._concentration0 = 1. + self._smoothness \
+        * (self._max - self._mod) / (self._max - self._min)
+
+      self._scale = self._max - self._min
+
+      # Broadcasting scale on affine bijector to match batch dimension.
+      # This prevents dimension mismatch for operations like cdf.
+      super(PERT, self).__init__(
+          distribution=Beta(
+              concentration1=self._concentration1,
+              concentration0=self._concentration0,
+              allow_nan_stats=allow_nan_stats),
+          bijector=AffineScalar(
+              shift=self._min,
+              scale=tf.broadcast_to(
+                  input=self._scale,
+                  shape=self._batch_shape_tensor(
+                      concentration1=self._concentration1,
+                      concentration0=self._concentration0))),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  def _batch_shape_tensor(self, concentration1=None, concentration0=None):
+    return prefer_static.broadcast_shape(
+        prefer_static.shape(
+            self.concentration1 if concentration1 is None else concentration1),
+        prefer_static.shape(
+            self.concentration0 if concentration0 is None else concentration0))
+
+  def _parameter_control_dependencies(self, is_init):
+    if not self.validate_args:
+      return []
+    assertions = []
+    if is_init != tensor_util.is_ref(self._mod):
+      if is_init != tensor_util.is_ref(self._min):
+        assertions.append(assert_util.assert_greater(
+            self._mod,
+            self._min,
+            message="Mode must be greater than minimum."
+        ))
+      if is_init != tensor_util.is_ref(self._max):
+        assertions.append(assert_util.assert_greater(
+            self._max,
+            self._mod,
+            message="Maximum must be greater than mode."
+        ))
+    if is_init != tensor_util.is_ref(self._smoothness):
+      assertions.append(assert_util.assert_positive(
+          self._smoothness,
+          message="Smoothness parameter must be positive."
+      ))
+    return assertions
+
+  def _survivial_function(self, value, **kwargs):
+    return 1. - self.cdf(value, kwargs)
```

Comment:
this should already be implemented as such in the base class (i.e. you can leave it out)